### DISCUSSION TITLE
Wemos D1 Recieve and Transmit, how?

### DISCUSSION THREAD

**Discussion by javrolijk** [timmbogner/Farm-Data-Relay-System#95] [Q&A] [answered]
Hi folks,

I have a question related to the picture below. I also have two Wemos D1 boards but not quite understanding how to send data between these two boards. Do I need to connect a wire on the TX on one board to RX to the other board?

This is a very great project, thanks Tim and others. 👍

Thanks in advance for your help.

Jairo

![Screenshot_20220814-184950_Chrome](https://user-images.githubusercontent.com/70583691/184557821-98e3ac65-83d6-4a2e-abf6-878ba9384a21.jpg)

**Comment by timmbogner** [comment, marked as answer]
Thanks!

Yep, you have it right.  You'll want to connect the TX pin of one device to the Rx pin of a second device,.

ESP8266 doesn't have multiple serial ports, so you should disable serial debugging messages (FDRS_DEBUG) on any devices that send serial data as well.

**Reply by javrolijk** [reply]
Thank you very much Tim. Works as it should, I will start a discussion on another topic.

**Comment by svdrummer** [comment]
Incase you have them both running on thier own batteries, you also need a
common ground/negative/0volt or whatever.

**Reply by dtuando** [reply]
Quick question why not use one board as the gateway. I have used one board in a similar project I've done in the past. It can communicate with esp-now using the AP Mac address and still connect to the internet via STA Mac address?

**Reply by timmbogner** [reply]
When an ESP connects to a WiFi access point, it switches the radio to the channel of the the AP. This also changes the channel that ESP-NOW is running on, but doesn't change the channel that the rest of the devices are on. Sometimes this doesn't cause problems, but often it does.
